### PR TITLE
[webgpu-native] opt matmulnbits

### PR DESCRIPTION
**Pull request by qjia7**

### CODE REVIEW

**Comment by qjia7** [comment]
All native cases can pass now. @guschmue @fs-eire Please take a look, thanks.

File: onnxruntime/contrib_ops/webgpu/quantization/matmul_nbits.cc [hunk 1]
```diff
@@ -53,167 +53,268 @@ ONNX_OPERATOR_KERNEL_EX(
...
+    shader.AdditionalImplementation() << "var<workgroup> sub_a: array<input_a_value_t, " << a_length_per_tile << ">;\n"
+                                      << "var<workgroup> inter_results: array<array<output_value_t, " << WorkgroupSizeX() << ">, " << WorkgroupSizeY() << ">;\n";
+    std::string offset = "workgroup_idx * " + std::to_string(WorkgroupSizeY());
+    shader.MainFunctionBody() << "  let output_indices = " << y.OffsetToIndices(offset) << ";\n"
+                              << "  let col = output_indices[2];\n"
+                                 "  let row = output_indices[1];\n"
+                                 "  let batch = output_indices[0];\n"
+                                 "  let n_blocks_per_col = uniforms.input_b_shape[1];\n"
+                              << "  let num_tiles =  (n_blocks_per_col - 1) / " << blocks_per_tile << " + 1;\n"
+                              << "  // Loop over shared dimension.\n"
```
Comment (fs-eire) [inline onnxruntime/contrib_ops/webgpu/quantization/matmul_nbits.cc:75]: maybe we can put comments outside?

Comment (qjia7) [reply onnxruntime/contrib_ops/webgpu/quantization/matmul_nbits.cc]: Done.

**Review by guschmue — APPROVED** [review]

**Comment by guschmue** [comment]
also tested and did some perf comparison - yeah, it's good on Xe: Phi3 on tlk token/sec went up from 8.5 -> 13.1

**Review by fs-eire — APPROVED** [review]

**Pull request merged**